assets/references.go

```diff
@@ -22,6 +22,12 @@ type Reference interface {
 	Variable() bool
 }
 
+// UUIDReference is interface for all reference types that contain a UUID
+type UUIDReference interface {
+	Reference
+	GenericUUID() utils.UUID
```

Comment:
having different types for every kind of UUID is mostly good but sometimes hurts